api/src/main/java/org/infinispan/api/mutiny/MutinyCache.java

```diff
@@ -51,132 +53,366 @@
 
    /**
     * Insert the key/value if such key does not exist
     *
     * @param key
     * @param value
-    * @return Void
+    * @return the previous value if present
+    */
+   default Uni<V> putIfAbsent(K key, V value) {
+      return putIfAbsent(key, value, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param key
+    * @param value
+    * @param options
+    * @return the previous value if present
     */
-   Uni<Boolean> putIfAbsent(K key, V value);
+   Uni<V> putIfAbsent(K key, V value, CacheWriteOperationOptions options);
+
+   /**
+    * Insert the key/value if such key does not exist
+    *
+    * @param key
+    * @param value
+    * @return
+    */
+   default Uni<Boolean> setIfAbsent(K key, V value) {
+      return setIfAbsent(key, value, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param key
+    * @param value
+    * @param options
+    * @return
+    */
+   Uni<Boolean> setIfAbsent(K key, V value, CacheWriteOperationOptions options);
 
    /**
     * Save the key/value. If the key exists will replace the value
     *
     * @param key
     * @param value
-    * @return Void
+    * @return
+    */
+   default Uni<V> put(K key, V value) {
+      return put(key, value, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param key
+    * @param value
+    * @param options
+    * @return
     */
-   Uni<V> put(K key, V value);
+   Uni<V> put(K key, V value, CacheWriteOperationOptions options);
 
    /**
     * @param key
     * @param value
     * @return
     */
-   Uni<Void> set(K key, V value);
+   default Uni<Void> set(K key, V value) {
+      return set(key, value, CacheWriteOperationOptions.DEFAULT);
+   }
+
+   /**
+    * @param key
+    * @param value
+    * @param options
+    * @return
+    */
+   Uni<Void> set(K key, V value, CacheWriteOperationOptions options);
+
+   /**
+    * Delete the key
+    *
+    * @param key
+    * @return true if the key existed and was removed, false if the key did not exist.
+    */
+   default Uni<Boolean> remove(K key) {
+      return remove(key, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Delete the key
     *
     * @param key
+    * @param options
     * @return true if the key existed and was removed, false if the key did not exist.
     */
-   Uni<Boolean> remove(K key);
+   Uni<Boolean> remove(K key, CacheOperationOptions options);
+
+   /**
+    * Removes the key and returns its value if present.
+    *
+    * @param key
+    * @return the value of the key before removal. Returns null if the key didn't exist.
+    */
+   default Uni<V> getAndRemove(K key) {
+      return getAndRemove(key, CacheOperationOptions.DEFAULT);
+   }
 
    /**
     * Removes the key and returns its value if present.
     *
     * @param key
+    * @param options
     * @return the value of the key before removal. Returns null if the key didn't exist.
     */
-   Uni<V> getAndRemove(K key);
+   Uni<V> getAndRemove(K key, CacheOperationOptions options);
```

Comment:
`getAndReplace` missing too

Reply:
we have `getOrReplaceEntry`: we discussed the fact that when replacement fails, you want to have the new version returned.